src/container/container.ts

```diff
@@ -620,9 +619,8 @@ class Container implements interfaces.Container {
                 args.isMultiInject,
                 args.targetType,
                 args.serviceIdentifier,
-                args.key,
-                args.value,
-                args.avoidConstraints
+                args.avoidConstraints,
+                args.tags
```

Comment:
This does not place nicely with existing middleware

```

        it.only("does not work with middleware", () => {
            const middleware:interfaces.Middleware = (next => {
                return (nextArgs => {
                    if(nextArgs.key === "tag" && nextArgs.value === 1){
                        nextArgs.value = 2;
                    }
                    return next(nextArgs);
                })
            })

            const container = new Container();
            container.applyMiddleware(middleware);
            container.bind('Sid').toConstantValue("One").whenTargetTagged("tag",1);
            container.bind('Sid').toConstantValue("Two").whenTargetTagged("tag",2);
            expect(container.getTagged<string>("Sid","tag",1)).to.equal("Two");
        })
```

Reply:
Oh, because changing key and value in the middleware does not affect the tags therefore the resolution is not affected. Good catch! I'll try to come around that.